braingeneers/utils/configure.py

```diff
@@ -42,6 +38,7 @@
         'pandas',
         'nptyping',
         'paho-mqtt',
+        'joblib>=1.3.0',
```

Comment:
Concerning your question about this dependency, this is in line with common practice, and is imo the most appropriate place to put this.  Dependencies are not specified at the function-level or class-level that I've ever seen.  If it is important (or cumbersome) enough, it might warrant creating a new extension, but this is fine here.

Suggestion: Pinning under version 2 can prevent accidental breaking changes, such as:

```
joblib>=1.3.0,<2
```